### PR TITLE
EVG-16219 Upgrade storybook and add vite support

### PR DESCRIPTION
**Pull request by khelif96**
[EVG-16219](https://jira.mongodb.org/browse/EVG-16219)

### Description 
* Upgrades storybook to the latest release
* Upgrades storybook deps to latest releases
* Migrates storybook to use existing vite config
* Reorganize stories
* Add automatic storybook doc generation

### Screenshots
image

### CODE REVIEW

File: src/components/SpruceForm/SpruceForm.stories.tsx [hunk 1]
```diff
@@ -81,13 +68,13 @@ const example1Def = {
     properties: {
```
Comment (minnakt) [inline src/components/SpruceForm/SpruceForm.stories.tsx:68]: For every `title` below this line, did you mean to add `Components/`? These are the titles within the form and so they look a little odd

Screen Shot 2022-06-29 at 1 17 40 PM

File: src/components/TreeSelect/TreeSelect.stories.tsx [hunk 2]
```diff
@@ -3,34 +3,34 @@ import { TreeSelect } from "components/TreeSelect";
 
 const treeData = [
```
Comment (minnakt) [inline src/components/TreeSelect/TreeSelect.stories.tsx:4]: Same question about having `Components/` in the `title` of `TreeSelect`

Screen Shot 2022-06-29 at 1 20 43 PM

File: src/components/WelcomeModal/WelcomeModal.stories.tsx [hunk 3]
```diff
@@ -37,8 +37,8 @@ export const MovieCard = () => (
   <MockedProvider>
     <CarouselCard
       card={{
-        title: "A Title",
-        subtitle: "A Subtitle",
+        title: "Components/A Title",
+        subtitle: "Components/A Subtitle",
```
Comment (minnakt) [inline src/components/WelcomeModal/WelcomeModal.stories.tsx:41]: Same question about `Components/` in title

Screen Shot 2022-06-29 at 1 21 24 PM

File: src/components/Icon/Icon.stories.tsx [hunk 4]
```diff
@@ -1,36 +1,40 @@
...
+
 export default {
-  title: "Spruce Icons",
-  decorators: [withKnobs],
+  title: "Components/Icons",
   component: Icon,
+  args: {
+    color: "#000000",
+    size: Sizes[Size.Default],
+    text: "Tooltip Text",
```
Comment (minnakt) [inline src/components/Icon/Icon.stories.tsx:18]: `Tooltip Text` might not belong here. I think it's only needed for `TooltipIcon`

File: src/components/Dropdown/Dropdown.stories.tsx [hunk 5]
```diff
@@ -1,23 +1,25 @@
...
-export const Story = () => (
-  <Dropdown disabled={boolean("disabled", false)} buttonText="Test">
-    Some Children
-  </Dropdown>
-);
+export const Story = (args) => <Dropdown {...args}>Some Children</Dropdown>;
+Story.args = {
+  disabled: false,
+  text: "Test",
+};
```
Comment (minnakt) [inline src/components/Dropdown/Dropdown.stories.tsx:15]: Do you need this and `CustomButtonRender.args` on `L23`? 🤔 I removed them and it seems like it still works the same way, with the disable functionality

File: .storybook/preview.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+      color: /(background|color)$/i,
+      date: /Date$/,
+    },
+  },
+  apolloClient: {
+    MockedProvider,
+  },
+};
+
+// @storybook/addon-storyshots isn't successfully wrapping the decorators from storybbook-addon-react-router-v6 and storybook-addon-apollo-client so we need to manually wrap the stories.
```
Comment (minnakt) [inline .storybook/preview.tsx:20]: nit.  typo `storybbook-addon-react-router-v6` => `storybook-addon-react-router-v6`

File: src/pages/task/taskTabs/buildBaronAndAnnotations/AnnotationTicketsTable/AnnotationTicketRow.stories.tsx [hunk 7]
```diff
@@ -1,13 +1,24 @@
...
+  args: {
+    confidenceScore: 0.5,
+  },
+  argTypes: {
+    confidenceScore: {
+      control: {
+        type: "range",
+        min: 0,
+        max: 1,
+        step: 0.01,
```
Comment (minnakt) [inline src/pages/task/taskTabs/buildBaronAndAnnotations/AnnotationTicketsTable/AnnotationTicketRow.stories.tsx:15]: this is really cool!

**Review by minnakt — COMMENTED** [review]
It looks really good! Just have a few questions and small comments 🐛

File: src/components/SpruceForm/SpruceForm.stories.tsx (hunk 1 above)
Comment (khelif96) [reply src/components/SpruceForm/SpruceForm.stories.tsx:68]: You have witnessed the limits of my find and replace skills.

File: .storybook/main.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+      react({
+        jsxImportSource: "@emotion/react",
+        babel: {
+          plugins: ["@emotion/babel-plugin", "import-graphql"],
+        },
+        fastRefresh: true,
+        // exclude story book stories from fast refresh (storybook should handle this itself)
+        exclude: [/\.stories\.([tj])sx?$/, /node_modules/],
+      })
+    );
```
[the remark this answers is not in the document]
Comment (khelif96) [reply .storybook/main.ts:65]: Storybook has its own version of this plugin. We remove both of them and then add the one we want.

File: .storybook/main.ts [hunk 9]
```diff
@@ -0,0 +1,70 @@
+import type { StorybookViteConfig } from "@storybook/builder-vite";
+import { mergeConfig } from "vite";
+// @ts-ignore
+import viteConfig from "../vite.config";
```
[the remark this answers is not in the document]
Comment (khelif96) [reply .storybook/main.ts]: Fixed there was a tsconfig change i needed to make

File: src/components/SearchableDropdown/SearchableDropdown.stories.tsx [hunk 10]
```diff
@@ -2,7 +2,7 @@ import { useState } from "react";
 import SearchableDropdown from ".";
 
 export default {
-  title: "Searchable Dropdown",
+  title: "Components/Searchable Dropdown",
   component: SearchableDropdown,
 };
```
Comment (minnakt) [inline src/components/SearchableDropdown/SearchableDropdown.stories.tsx:8]: nit. Could you change the `Story` to `Default` in this one too?

Screen Shot 2022-07-01 at 12 02 31 PM

**Review by minnakt — APPROVED** [review]
Looks good to me! 💚🐛

Just left a few comments related to a few visual fixes that would be nice to have!

**Pull request merged**